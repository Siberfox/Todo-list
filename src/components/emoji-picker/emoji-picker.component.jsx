import React from "react";

import "./emoji-picker.styles.scss";

export const emoji = [
  { id: 1, label: "Document", sign: "📝" },
  { id: 2, label: "Bed", sign: "🛌" },
  { id: 3, label: "Time", sign: "⌚" },
  { id: 4, label: "Knife", sign: "🔪" },
  { id: 5, label: "Bomb", sign: "💣" },
  { id: 6, label: "Holiday", sign: "🎉" },
  { id: 7, label: "Gift", sign: "🎁" },
  { id: 8, label: "Phone", sign: "📞" },
  { id: 9, label: "Money", sign: "💰" },
  { id: 10, label: "Shop cart", sign: "🛒" },
];

export const EmojiPicker = ({ addEmoji }) => {
  return (
    <div className="emoji-picker">
      {emoji.map((item) => (
        <span
          className="emoji-item"
          onClick={addEmoji}
          role="img"
          aria-label={item.label}
          key={item.id}
        >
          {item.sign}
        </span>
      ))}
    </div>
  );
};
