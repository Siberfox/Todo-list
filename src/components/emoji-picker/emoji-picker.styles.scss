.emoji-picker {
  position: absolute;
  top: 35px;
  height: 30px;
}
.emoji-item {
  font-size: 22px;
  margin-right: 6px;
}
