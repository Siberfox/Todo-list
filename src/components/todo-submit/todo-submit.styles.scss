.todo-submit {
  display: flex;
  justify-content: space-between;
  margin: 30px auto;
  position: relative;
  &_input {
    width: 300px;
    height: 30px;
    padding-left: 35px;
    border-radius: 4px;
  }
  &_button {
    margin-left: 10px;
    border-radius: 4px;
    background-color: #6fcf97;
    &:hover {
      background-color: darken(#6fcf97, 20%);
    }
    &:active {
      background-color: #6fcf97;
    }
  }
  .emoji-icon {
    position: absolute;
    margin-left: 7px;
    margin-top: 8px;
  }
}
