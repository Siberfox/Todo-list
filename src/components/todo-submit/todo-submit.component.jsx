import React from "react";

import "./todo-submit.styles.scss";

import "emoji-mart/css/emoji-mart.css";
import { ReactComponent as EmojiIcon } from "../../assets/emoji.svg";
import { EmojiPicker } from "../emoji-picker/emoji-picker.component";

export const TodoSubmit = ({
  addItem,
  inputValue,
  handleChange,
  addEmoji,
  emoji,
  hidden,
  toggleEmojiHidden,
}) => {
  return (
    <form onSubmit={addItem} className="todo-submit">
      <input
        className="todo-submit_input"
        type="text"
        name="text"
        value={inputValue}
        onChange={handleChange}
        placeholder="add todo..."
        required
      />
      <div className="emoji-icon" onClick={toggleEmojiHidden}>
        <EmojiIcon />
      </div>
      {hidden ? null : <EmojiPicker addEmoji={addEmoji} emoji={emoji} />}
      <button type="submit" className="todo-submit_button">
        Add
      </button>
    </form>
  );
};
