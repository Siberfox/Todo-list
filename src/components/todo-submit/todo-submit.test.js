import React from "react";
import { shallow } from "enzyme";

import { TodoSubmit } from "./todo-submit.component";
import { EmojiPicker } from "../emoji-picker/emoji-picker.component";

describe("todo submit", () => {
  const mockAddItem = jest.fn();
  const mockHandleChange = jest.fn();
  const mockAddEmoji = jest.fn();
  const mockToggleEmojiHidden = jest.fn();

  const mockProps = {
    addItem: mockAddItem,
    inputValue: "todo",
    handleChange: mockHandleChange,
    addEmoji: mockAddEmoji,
    hidden: false,
    toggleEmojiHidden: mockToggleEmojiHidden,
  };

  const wrapper = shallow(<TodoSubmit {...mockProps} />);

  it("should render component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("input field value should be props inputValue", () => {
    expect(wrapper.find("input").prop("value")).toEqual("todo");
  });

  it("should call handleChange on change", () => {
    wrapper.find("input").simulate("change");
    expect(mockHandleChange).toHaveBeenCalled();
  });

  it("should call addItem on form submit", () => {
    wrapper.find("form").simulate("submit");
    expect(mockAddItem).toHaveBeenCalled();
  });

  it("show emoji picker when prop hidden is false", () => {
    expect(wrapper.exists(EmojiPicker)).toBe(true);
  });

  it("hidden emoji picker when prop hidden is true", () => {
    const mockHidden = true;
    const wrapper = shallow(<TodoSubmit hidden={mockHidden} />);

    expect(wrapper.exists(EmojiPicker)).toBe(false);
  });

  it("should call toggleEmojiHidden on click emoji icon", () => {
    wrapper.find(".emoji-icon").simulate("click");
    expect(mockToggleEmojiHidden).toHaveBeenCalled();
  });
});
