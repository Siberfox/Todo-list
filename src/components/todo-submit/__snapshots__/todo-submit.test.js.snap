// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo submit should render component 1`] = `
<form
  className="todo-submit"
  onSubmit={[MockFunction]}
>
  <input
    className="todo-submit_input"
    name="text"
    onChange={[MockFunction]}
    placeholder="add todo..."
    required={true}
    type="text"
    value="todo"
  />
  <div
    className="emoji-icon"
    onClick={[MockFunction]}
  >
    <ForwardRef(SvgEmoji) />
  </div>
  <EmojiPicker
    addEmoji={[MockFunction]}
  />
  <button
    className="todo-submit_button"
    type="submit"
  >
    Add
  </button>
</form>
`;
